### PR TITLE
Support arm64 ins, aiming for 1B connections

### PR DESCRIPTION
**Pull request by qzhuyan**

### CODE REVIEW

File: user_data/emqx_init.sh [hunk 1]
```diff
@@ -126,6 +126,9 @@ rate_limit {
   conn_bytes_in = infinity
 }
 
+listeners.tcp.default {
+  max_connections = 1000000000
```
Comment (qzhuyan) [inline user_data/emqx_init.sh]: should be `infinity`

File: cdk_emqx_cluster/cdk_emqx_cluster_stack.py [hunk 2]
```diff
@@ -33,6 +33,9 @@
     "eu-west-1": "ami-08edbb0e85d6a0a07",  # ubuntu 20.04 latest
 })
 
+ubuntu_arm_ami = ec2.MachineImage.from_ssm_parameter('/aws/service/canonical/ubuntu/server/focal/stable/current/arm64/hvm/ebs-gp2/ami-id')
+ubuntu_x86_64_ami = ec2.MachineImage.from_ssm_parameter('/aws/service/canonical/ubuntu/server/focal/stable/current/arm64/hvm/ebs-gp2/ami-id')
```
Comment (thalesmg) [inline cdk_emqx_cluster/cdk_emqx_cluster_stack.py]: Looks like both SSM parameters are the same (arm64)?

File: user_data/emqx_init.sh (hunk 1 above)
Comment (thalesmg) [inline user_data/emqx_init.sh]: I believe there's a line for this config [above](https://github.com/emqx/cdk-emqx-cluster/blob/f853cde2d02079aa2212fcf8363d40a9209bcd5a/user_data/emqx_init.sh#L74).

**Review by thalesmg — APPROVED** [review]

File: user_data/os_common.sh [hunk 3]
```diff
@@ -1,15 +1,15 @@
 #!/bin/bash
 
 sudo bash -c 'echo "session required pam_limits.so" >> /etc/pam.d/common-session'
-sudo bash -c 'echo "*      soft    nofile      2000000"  >> /etc/security/limits.conf'
-sudo bash -c 'echo "*      hard    nofile      2000000"  >> /etc/security/limits.conf'
+sudo bash -c 'echo "*      soft    nofile      100000000"  >> /etc/security/limits.conf'
+sudo bash -c 'echo "*      hard    nofile      100000000"  >> /etc/security/limits.conf'
 
 echo "net.ipv4.tcp_tw_reuse=1" >>  /etc/sysctl.d/99-sysctl.conf
 echo "fs.nr_open=8000000" >>  /etc/sysctl.d/99-sysctl.conf
```
Comment (thalesmg) [inline user_data/os_common.sh]: Q: Should this be `1000000000` as well?

Comment (thalesmg) [reply user_data/os_common.sh]: I'd suggest we put this in a variable, so we can just use it on those occurrences.

File: cdk_emqx_cluster/cdk_emqx_cluster_stack.py (hunk 2 above)
Comment (qzhuyan) [reply cdk_emqx_cluster/cdk_emqx_cluster_stack.py]: yes, corrected.

File: user_data/os_common.sh (hunk 3 above)
Comment (qzhuyan) [reply user_data/os_common.sh]: right, it should be some systemctl command to reload it.

File: cdk_emqx_cluster/cdk_emqx_cluster_stack.py [hunk 4]
```diff
@@ -33,6 +33,9 @@
     "eu-west-1": "ami-08edbb0e85d6a0a07",  # ubuntu 20.04 latest
 })
 
+ubuntu_arm_ami = ec2.MachineImage.from_ssm_parameter('/aws/service/canonical/ubuntu/server/focal/stable/current/amd64/hvm/ebs-gp2/ami-id')
```
Comment (thalesmg) [inline cdk_emqx_cluster/cdk_emqx_cluster_stack.py]: I believe they are switched now: `ubuntu_arm_ami` points to `/amd64/` and `ubuntu_x86_64_ami` points to `/arm64/`

File: user_data/os_common.sh (hunk 3 above)
Comment (qzhuyan) [reply user_data/os_common.sh]: Now I tried and remember both `systemd` and `sysctl -p` refused to set the value I want.

**Review by thalesmg — APPROVED** [review]
Micro nit: it seems that [some scripts](https://github.com/emqx/cdk-emqx-cluster/pull/32/files#diff-cc175b7a4311ffa4846b1e0a9eebeb1c803497bce0eed61745f48b87cc91612cR213) are using 8 M file descriptor limit. (not sure if intentional)

**Pull request merged**